src/app/debug/filter-side-drawer/filter.service.ts

```diff
@@ -42,4 +70,31 @@ export class FilterService {
   setCurrentRecords(records: Map<string, Array<string>>): void {
     this.currentRecordsSubject.next(records);
   }
+
+  setMetadataTypes(metadataTypes: Map<string, string>): void {
+    this.metadataTypes = new Map<string, string>(Object.entries(metadataTypes));
+    this.metadataTypesSubject.next(this.metadataTypes);
+  }
+
+  disableFilterError(): void {
+    this.filterErrors.clear();
+    this.filterErrorSubject.next([false, this.filterErrors]);
+  }
+
+  enableFilterError(inputFilter: Map<string, string>): void {
+    this.filterErrors = inputFilter;
+    this.filterErrorSubject.next([true, this.filterErrors]);
+  }
+
+  isInValidTimestamp(userInput: string, metadataName: string): boolean {
+    return this.metadataTypes.get(metadataName) == 'timestamp' && Number.isNaN(Date.parse(userInput));
+  }
+
+  isInValidNumber(userInput: string, metadataName: string): boolean {
+    const regex: RegExp = /^-?\d+(\.\d+)?$/;
+    return (
+      (this.metadataTypes.get(metadataName) == 'int' || this.metadataTypes.get(metadataName) == 'long') &&
+      (Number.isNaN(Number.parseFloat(userInput)) || !regex.test(userInput))
+    );
```

Comment:
```suggestion
    const metadataType = this.metadataTypes.get(metadataName);
    return (
      (type == 'int' || type == 'long') &&
      (Number.isNaN(Number.parseFloat(userInput)) || !regex.test(userInput))
    );
```

Reply:
I would like long `if` statements like these to be encapsulated.

Reply:
Changed it in the new commit